RedHatInsights/insights-inventory-frontend
# fix(ESSNTL-4731): Detect 404 from Hosts

[ESSNTL-4731](https://issues.redhat.com/browse/ESSNTL-4731)

When an user visited a non existing page (eg. had 20 systems and displayed 20 per page, the second page would throw a 404) he would be faced with an error page.

Now the user is redirected to the first page and gets an error notification.

Had to use `useSearchParams` because there's an edge case when tags finish fetching first and page is set to 1 ignoring the url params.

How to test:
1. Go to Inventory Systems and change the page to a page number that doesn't exist
2. For some reason it could take several tries to get the error message. (Could be a router upgrade regression?)
3. After the error message you should be redirected to the first page and the table should correctly load.
